Failing Unit Test javascript/position_caret/test_position_caret

Running py35 on current master (50fb4a78c7ac2d106c48e966b9f1f9a2b7831dd1) gives a unit test failure, specific test output below.

Machine is an ArchLinux x86_64, updated yesterday (Sep 28), no testing repo.

```
tox -e py35 -- tests/unit/javascript/position_caret/test_position_caret.py 
py35 installed: beautifulsoup4==4.5.1,CherryPy==8.1.0,click==6.6,colorama==0.3.7,coverage==4.2,cssutils==1.0.1,decorator==4.0.10,Flask==0.11.1,glob2==0.4.1,httpbin==0.5.0,hypothesis==3.5.2,itsdangerous==0.24,Jinja2==2.8,Mako==1.0.4,MarkupSafe==0.23,parse==1.6.6,parse-type==0.3.4,py==1.4.31,Pygments==2.1.3,pyPEG2==2.15.2,pytest==3.0.2,pytest-bdd==2.18.0,pytest-catchlog==1.2.2,pytest-cov==2.3.1,pytest-faulthandler==1.3.0,pytest-instafail==0.3.0,pytest-mock==1.2,pytest-qt==2.0.0,pytest-repeat==0.4.1,pytest-rerunfailures==2.0.1,pytest-travis-fold==1.2.0,pytest-warnings==0.1.0,pytest-xvfb==0.3.0,PyYAML==3.12,six==1.10.0,spark-parser==1.4.0,uncompyle6==2.8.3,vulture==0.10,Werkzeug==0.11.11,xdis==2.3.1
py35 runtests: PYTHONHASHSEED='3175465611'
py35 runtests: commands[0] | /home/dk/projects/extern/qutebrowser/.tox/py35/bin/python scripts/link_pyqt.py --tox /home/dk/projects/extern/qutebrowser/.tox/py35
py35 runtests: commands[1] | /home/dk/projects/extern/qutebrowser/.tox/py35/bin/python scripts/dev/run_pytest.py tests/unit/javascript/position_caret/test_position_caret.py
========================================== test session starts ==========================================
platform linux -- Python 3.5.2, pytest-3.0.2, py-1.4.31, pluggy-0.3.1
PyQt5 5.7 -- Qt runtime 5.7.0 -- Qt compiled 5.7.0
rootdir: /home/dk/projects/extern/qutebrowser, inifile: pytest.ini
plugins: bdd-2.18.0, rerunfailures-2.0.1, warnings-0.1.0, repeat-0.4.1, mock-1.2, qt-2.0.0, faulthandler-1.3.0, cov-2.3.1, hypothesis-3.5.2, catchlog-1.2.2, xvfb-0.3.0, travis-fold-1.2.0, instafail-0.3.0
collected 5 items 

tests/unit/javascript/position_caret/test_position_caret.py ....F

________________________________________ test_scrolled_down_img _________________________________________

caret_tester = <test_position_caret.CaretTester object at 0x7f1f88d049b0>

    @pytest.mark.integration
    def test_scrolled_down_img(caret_tester):
        """Test with an image at the top with the viewport scrolled down."""
        caret_tester.js.load('position_caret/scrolled_down_img.html')
        caret_tester.js.scroll_anchor('anchor')
        caret_tester.check_scrolled()
>       caret_tester.check()

tests/unit/javascript/position_caret/test_position_caret.py:103: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <test_position_caret.CaretTester object at 0x7f1f88d049b0>

    def check(self):
        """Check whether the caret is before the MARKER text."""
        self.js.run_file('position_caret.js')
        self.js.webview.triggerPageAction(QWebPage.SelectNextWord)
>       assert self.js.webview.selectedText().rstrip() == "MARKER"
E       assert '' == 'MARKER'
E         + MARKER

tests/unit/javascript/position_caret/test_position_caret.py:57: AssertionError

======================================== short test summary info ========================================
FAIL tests/unit/javascript/position_caret/test_position_caret.py::test_scrolled_down_img
========================================== 0 tests deselected ===========================================
================================== 1 failed, 4 passed in 0.26 seconds ===================================
ERROR: InvocationError: '/home/dk/projects/extern/qutebrowser/.tox/py35/bin/python scripts/dev/run_pytest.py tests/unit/javascript/position_caret/test_position_caret.py'
________________________________________________ summary ________________________________________________
ERROR:   py35: commands failed
```

--- a/tests/unit/javascript/position_caret/test_position_caret.py
+++ b/tests/unit/javascript/position_caret/test_position_caret.py
@@ -67,7 +67,11 @@ class CaretTester:
 @pytest.fixture
 def caret_tester(js_tester):
     """Helper fixture to test caret browsing positions."""
-    return CaretTester(js_tester)
+    caret_tester = CaretTester(js_tester)
+    # Showing webview here is necessary for test_scrolled_down_img to
+    # succeed in some cases, see #1988
+    caret_tester.js.webview.show()
+    return caret_tester
 
 
 @pytest.mark.integration

